library/data/registry_entry_attachments/src/main/java/org/quiltmc/qsl/registry/attachment/impl/RegistryEntryAttachmentHolder.java

```diff
@@ -93,21 +94,45 @@ public static <R> RegistryEntryAttachmentHolder<R> getAssets(Registry<R> registr
 	}
 
 	public final Table<RegistryEntryAttachment<R, ?>, R, Object> valueTable;
+	public final Table<RegistryEntryAttachment<R, ?>, TagKey<R>, Object> valueTagTable;
 
 	@SuppressWarnings("UnstableApiUsage")
 	private RegistryEntryAttachmentHolder() {
 		this.valueTable = Tables.newCustomTable(new Object2ReferenceOpenHashMap<>(), Reference2ObjectOpenHashMap::new);
+		this.valueTagTable = Tables.newCustomTable(new Object2ReferenceOpenHashMap<>(), Reference2ObjectOpenHashMap::new);
 	}
 
 	@SuppressWarnings("unchecked")
 	public <V> V getValue(RegistryEntryAttachment<R, V> attachment, R entry) {
-		return (V) this.valueTable.get(attachment, entry);
+		V value = (V) this.valueTable.get(attachment, entry);
+		if (value == null) {
```

Comment:
This method is pretty non-trivial; some documentation on the expected behavior of how this method resolves a value using tags would be nice.